### PR TITLE
[WIP]SLE-21378: adds USBGuard content

### PR DESCRIPTION
**Pull request by Amrita42**
### PR creator: Description

Describe the overall goals of this pull request.
Document USBGuard

### PR creator: Are there any relevant issues/feature requests?

[Jira](https://jira.suse.com/login.jsp?os_destination=%2Fbrowse%2FSLE-21378)

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [X] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [X] SLE 15 SP4/openSUSE Leap 15.5
  - [X] SLE 15 SP4/openSUSE Leap 15.4
  - [ ] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by rfrohl** [comment]
> Robert, One query when I install usbguard tools I get the following: 15.5 The following 4 NEW packages are going to be installed: libqb100 libusbguard1 usbguard usbguard-tools
> 
> I have deleted my 15.4 instance , is it the same ? Thanks Amrita

Sorry only saw the question now, please use @rfrohl to get a quicker response. 

I do not think the dependencies should be documented in general, as the might change based on build dependencies or other changes. 

Checked a Leap 15.4 instance, which is using the same packages as SLE 15-SP4:

```
$ cat /etc/os-release 
NAME="openSUSE Leap"
VERSION="15.4"
ID="opensuse-leap"
ID_LIKE="suse opensuse"
VERSION_ID="15.4"
PRETTY_NAME="openSUSE Leap 15.4"
[..]
$ zypper in usbguard-tools
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 6 NEW packages are going to be installed:
  libprotobuf20 libqb100 libsodium23 libusbguard1 usbguard usbguard-tools

6 new packages to install.
Overall download size: 1,6 MiB. Already cached: 0 B. After the operation, additional 5,4 MiB will be used.
Continue? [y/n/v/...? shows all options] (y):
```

**Comment by Amrita42** [comment]
@rfrohl ok noted. Can you please take a look at the rest of the content and provide a tech review ? Thanks

File: xml/physical_security.xml [hunk 1]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
      </step>
     </procedure>
    </sect1>
+  <sect1 xml:id="sec-USBGuard">
+<title>System protection with <emphasis>USBGuard against rogue USB devices</emphasis></title>
```
Comment (rfrohl) [inline xml/physical_security.xml:430]: Personally I find 'rogue USB devices' sounds to much like marketing and not really technical.. Maybe using 'USB security attacks' is another option [0] or something similar to 'intrusive USB devices' [1]. 

The man page [2] talks about 'enforces the USB device authorization', maybe a third option would be to use 'System protection by enforcing USB device authorization'.

What ever works in the big picture, I would advise against using 'rogue'.
[0] https://www.redhat.com/en/blog/built-protection-against-usb-security-attacks-usbguard
[1] https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-using-usbguard
[2] https://man.archlinux.org/man/usbguard-daemon.8.en

File: xml/physical_security.xml [hunk 2]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
...
+   <para>The rule language for writing USB device authorization policies</para>
+  </listitem>
+  <listitem><para>The C++ API for interacting with the daemon component implemented in a shared library</para></listitem>
+   </itemizedlist>
+   <sect2 xml:id="sec-installing-usbguard">
+    <title>Installing USBGuard</title>
+    <para>You can install and configure USBGuard.</para>
+    <procedure>
+      <step><para>To install USBGuard:</para>
+      <screen>&prompt.sudo; zypper install usbguard-tools</screen>
```
Comment (rfrohl) [inline xml/physical_security.xml:456]: From a technical point: `zypper install usbguard` should be enough to get a working daemon.

usbguard-tools is only needed for the CLI command that allows the user to interact with the daemon in an easier fashion, like generating policy or allowing/blocking a usb device during runtime (i.e. not adding a persitant rule to the policy that would survive reboots). usbguard-tools is not strictly needed though, one could write the rules by hand or get the rules from another host, identical host system.

File: xml/physical_security.xml [hunk 3]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
...
+              </listitem>
+                </varlistentry>
+   </variablelist>
+   <example xml:id="usbguard-config-file">
+    <title>USBGuard configuration file</title>
+ <screen>RuleFile=/etc/usbguard/rules.conf
+  IPCAccessControlFiles=/etc/usbguard/IPCAccessControl.d/</screen>
+  <para>The example tells the USBGuard daemon to load the rules from
+    <filename>/etc/usbguard/rules.conf</filename> file and allow users from the
+    <emphasis>usbguard</emphasis> group to access the IPC interface.</para>
```
[the remark this answers is not in the document]
Comment (rfrohl) [reply xml/physical_security.xml:561]: If this is true, then please note that `wheel` is not consistently used in SLE. It might even not exist at all for some service packs.

File: xml/physical_security.xml (hunk 1 above)
Comment (Amrita42) [reply xml/physical_security.xml:430]: Sounds good , I will re-phrase

File: xml/physical_security.xml [hunk 4]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
      </step>
     </procedure>
    </sect1>
+  <sect1 xml:id="sec-USBGuard">
+<title>System protection with <emphasis>USBGuard against rogue USB devices</emphasis></title>
+<para>
+  The USBGuard software framework helps to protect your system against rogue USB devices.
+  It does this by implementing basic whitelisting and blacklisting capabilities based on the device attributes.
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:433]: Implemented

File: xml/physical_security.xml [hunk 5]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
...
+      <screen>&prompt.sudo; zypper install usbguard-tools</screen>
+    <para>Two packages are installed; <emphasis>usbguard</emphasis> and <emphasis>usbguard-tools</emphasis>.</para>
+    </step>
+    <step><para>To create a default rule set, switch to &rootuser; </para>
+      <screen>&prompt.root;usbguard generate-policy > /etc/usbguard/rules.conf</screen>
+      <note><para>You can customize USBGuard by editing <filename>/etc/usbguard/rules.conf</filename>
+      file.
+      </para></note>
+    </step>
+    <step><para> You can start the USBGuard daemon or ensure automatic enablement at
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:465]: I would leave it as it it :
ref https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/7.4_release_notes/new_features_hardware_enablement  (ex-red hatter :)  cant help it )
https://www.ibm.com/docs/en/sgklm/3.0.1?topic=steps-automatic-services-enablement

File: xml/physical_security.xml [hunk 6]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
...
+      system start by switching to &rootuser;:</para>
+      <screen>&prompt.root;systemctl start usbguard.service</screen>
+      <screen>&prompt.root;systemctl enable usbguard.service</screen>
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:468]: Implemented

File: xml/physical_security.xml [hunk 7]
```diff
@@ -426,4 +426,180 @@ polkit.addRule(function(action, subject) {
...
+   <example xml:id="usbguard-config-file">
+    <title>USBGuard configuration file</title>
+ <screen>RuleFile=/etc/usbguard/rules.conf
+  IPCAccessControlFiles=/etc/usbguard/IPCAccessControl.d/</screen>
+  <para>The example tells the USBGuard daemon to load the rules from
+    <filename>/etc/usbguard/rules.conf</filename> file and allow users from the
+    <emphasis>usbguard</emphasis> group to access the IPC interface.</para>
+   </example>
+   <example xml:id="usbguard-config-file-ipcaccess">
+    <title>Configuration allowing full IPC access to users <literal>root</literal>,<literal>Amy</literal>,and
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:564]: ok changed to Joe

File: xml/physical_security.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:561]: Yes , users root , joe and group wheel .
So if wheel is not used consistently , then any suggestion as to the group and how i should proceed here?

[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:561]: Attached is the pdf , ptal example 9.1 , so this is for illustrative purposes. 
[book-security_en.pdf](https://github.com/SUSE/doc-sle/files/13204186/book-security_en.pdf)

[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:561]: @rfrohl ptal

**Comment by Amrita42** [comment]
@rfrohl 
Attached pdf , except the last unresolved comment:
[book-security_en.pdf](https://github.com/SUSE/doc-sle/files/13204329/book-security_en.pdf)

File: xml/physical_security.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (rfrohl) [reply xml/physical_security.xml:563]: so there is something off with the example:

1) the indentation is mixed for 9.1
2) the text is talking about both examples it think, 9.1 and 9.2. But that is confusing. Alternatively 9.2 is missing the explanation text
3) The text 'and allow users from the _usbguard group_ to access the IPC interface.' is not correct I think.

To solve this I would drop 9.1 completely TBH, I do not think there is a lot of value in that example. I would not expect users to change this under normal circumstances. 9.2 is what most users will have to interact with at one point. So I would leave that in.

Not sure if there are other considerations behind adding 9.1. Hope this was helpful.

File: xml/physical_security.xml [hunk 8]
```diff
@@ -426,4 +426,182 @@ polkit.addRule(function(action, subject) {
...
+  </listitem>
+  <listitem><para>The C++ API for interacting with the daemon component implemented in a shared library</para></listitem>
+   </itemizedlist>
+   <sect2 xml:id="sec-installing-usbguard">
+    <title>Installing USBGuard</title>
+    <para>You can install and configure USBGuard.</para>
+    <procedure>
+      <step><para>To install USBGuard:</para>
+      <screen>&prompt.sudo; zypper install usbguard</screen>
+    <para>Two packages are installed; <emphasis>usbguard</emphasis> and <emphasis>usbguard-tools</emphasis>.</para>
```
Comment (rfrohl) [inline xml/physical_security.xml:457]: This section in now incorrect: usbguard-tools is not a hard dependency, so it will not be installed automatically too.

Maybe rephrase to:
'<emphasis>usbguard</emphasis> and the required dependencies will be installed, if the user needs to interact with the usbguard service then optionally <emphasis>usbguard-tools</emphasis> can be installed too.'

I personally would not make references towards how many packages are installed, because this might change during packaging later on. For example in a later service pack and then might make the documention look outdated to users looking at it. These changes would be probably also very hard to find for the doc team.

File: xml/physical_security.xml [hunk 9]
```diff
@@ -426,4 +426,182 @@ polkit.addRule(function(action, subject) {
...
+   <example xml:id="usbguard-config-file">
+    <title>USBGuard configuration file</title>
+ <screen>RuleFile=/etc/usbguard/rules.conf
+  IPCAccessControlFiles=/etc/usbguard/IPCAccessControl.d/</screen>
+  <para>The example tells the USBGuard daemon to load the rules from
+    <filename>/etc/usbguard/rules.conf</filename> file and allow users from the
+    <emphasis>usbguard</emphasis> group to access the IPC interface.</para>
+   </example>
+   <example xml:id="usbguard-config-file-ipcaccess">
+    <title>Configuration allowing full IPC access to users <literal>root</literal>,<literal>joe</literal>,and
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:566]: removed

File: xml/physical_security.xml (hunk 8 above)
Comment (Amrita42) [reply xml/physical_security.xml:457]: Removed and makes sense to not mention spec packages

Comment (Amrita42) [reply xml/physical_security.xml:457]: Re-phrased.

File: xml/physical_security.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/physical_security.xml:563]: Ok I will drop example 9.1 and retain 92 with an explanatory text.

**Comment by Amrita42** [comment]
@dariavladykina please review thanks.

File: xml/physical_security.xml [hunk 10]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+    interface for dynamic interaction and policy enforcement</para>
+  </listitem>
+  <listitem>
+   <para>The rule language for writing USB device authorization policies</para>
+  </listitem>
+  <listitem><para>The C++ API for interacting with the daemon component implemented in a shared library</para></listitem>
+   </itemizedlist>
+   <sect2 xml:id="sec-installing-usbguard">
+    <title>Installing USBGuard</title>
+    <para>You can install and configure USBGuard.</para>
```
Comment (dariavladykina) [inline xml/physical_security.xml:453]: This looks a bit incomplete, maybe add more description here?

File: xml/physical_security.xml [hunk 11]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+    <procedure>
+      <step><para>To install USBGuard:</para>
+      <screen>&prompt.sudo; zypper install usbguard</screen>
+    <para>USBGuard and the required dependencies are installed. If you want to interact with the USBGuard service, you can
+      install <literal>usbguard-tools</literal>.</para>
+    </step>
+    <step><para>To generate a rule set based on currently connected USB devices, switch to &rootuser; </para>
+      <screen>&prompt.root;usbguard generate-policy > /etc/usbguard/rules.conf</screen>
+      <note><para>You can customize USBGuard by editing <filename>/etc/usbguard/rules.conf</filename>
```
Comment (dariavladykina) [inline xml/physical_security.xml:462]:
```suggestion
      <note><para>You can customize USBGuard by editing the <filename>/etc/usbguard/rules.conf</filename>
```

File: xml/physical_security.xml [hunk 12]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+   <sect2 xml:id="sec-installing-usbguard">
+    <title>Installing USBGuard</title>
+    <para>You can install and configure USBGuard.</para>
+    <procedure>
+      <step><para>To install USBGuard:</para>
+      <screen>&prompt.sudo; zypper install usbguard</screen>
+    <para>USBGuard and the required dependencies are installed. If you want to interact with the USBGuard service, you can
+      install <literal>usbguard-tools</literal>.</para>
+    </step>
+    <step><para>To generate a rule set based on currently connected USB devices, switch to &rootuser; </para>
```
Comment (dariavladykina) [inline xml/physical_security.xml:460]:
```suggestion
    <step><para>To generate a rule set based on currently connected USB devices, switch to &rootuser;:</para>
```

File: xml/physical_security.xml [hunk 13]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+      file.
+      </para></note>
+    </step>
+    <step><para> You can start the USBGuard daemon or ensure automatic enablement at
+      system start by switching to &rootuser;:</para>
+      <screen>&prompt.root;systemctl enable --now usbguard.service</screen>
+        </step>
+    <step><para> You can start either authorize or deauthorize a device from interacting with the system. Note that
+      this depends on the value of the <literal>ImplicitPolicyTarget</literal> option in the <filename>usbguard-daemon.conf
+      </filename> file. This option is used to treat devices that don’t match any rule in the policy
```
Comment (dariavladykina) [inline xml/physical_security.xml:472]:
```suggestion
    <step><para> You can authorize or deauthorize a device from interacting with the system. Note that
      this depends on the value of the <literal>ImplicitPolicyTarget</literal> option in the <filename>usbguard-daemon.conf
      </filename> file. This option is used to treat devices that do not match any rule in the policy.
```

File: xml/physical_security.xml [hunk 14]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+    </para>
+      <screen>usbguard allow-device 6</screen>
+      <screen>usbguard block-device 6</screen>
+      <para> You can also use the <command>reject-device</command>option to deauthorize and
+        remove a device from the system.
+              </para>
+      <note><para>You can use the <command>usbguard --help </command> to see all the options.</para></note>
```
Comment (dariavladykina) [inline xml/physical_security.xml:479]:
```suggestion
      <note><para>Use the <command>usbguard --help </command> to see all the options.</para></note>
```

File: xml/physical_security.xml [hunk 15]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+    </step>
+    </procedure>
+   </sect2>
+<sect2 xml:id="sec-how-to-use-usbguard">
+<title>How to use USBGuard</title>
+<para>You can configure a security policy to protect your system with enforced USB device authorization
+  by implementing <emphasis>allow</emphasis> and <emphasis>block</emphasis> lists based on the device attributes.
+</para>
+<sect3><title>The USBGuard configuration file </title>
```
Comment (dariavladykina) [inline xml/physical_security.xml:488]: This looks like a lonely subsection, please remove it and rephrase if needed.

File: xml/physical_security.xml [hunk 16]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+  <para>The USBGuard daemon loads the <filename>usbguard-daemon.conf</filename> file after the command-line
+  options are parsed and are used to configure the run time parameters of the daemon. The file is by default, located at
+<filename>/etc/usbguard/usbguard-daemon.conf</filename>. Some of the options in the file include:</para>
```
Comment (dariavladykina) [inline xml/physical_security.xml:491]:
```suggestion
  <para>The USBGuard daemon loads the <filename>usbguard-daemon.conf</filename> file after the command-line
  options are parsed and are used to configure the runtime parameters of the daemon. The file is by default located at
<filename>/etc/usbguard/usbguard-daemon.conf</filename>. Some options in the file include:</para>
```

File: xml/physical_security.xml [hunk 17]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+    <term><option>RuleFile=<replaceable>PATH</replaceable></option></term>
+    <listitem>
+    <para>The USBGuard daemon uses this file to load the policy rule set from it and to write
+      new rules received through the IPC (inter-process communication) interface.
+      The default is <filename>%sysconfdir%/usbguard/rules.conf</filename>.</para> </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><option>ImplicitPolicyTarget= <replaceable>TARGET</replaceable></option></term>
+   <listitem>
+    <para> How to treat devices that do not match any rule in the policy for example, </para>
```
Comment (dariavladykina) [inline xml/physical_security.xml:504]:
```suggestion
    <para> How to treat devices that do not match any rule in the policy, for example:</para>
```

File: xml/physical_security.xml [hunk 18]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+              <para>Path to files that are interpreted by the daemon as IPC access control definition files.</para>
+              </listitem>
+                </varlistentry>
+   </variablelist>
+      <example xml:id="usbguard-config-file-ipcaccess">
+    <title>Configuration</title>
+ <screen>IPCAllowedUsers=root joe
+IPCAllowedGroups=wheel</screen>
+<para>The example allows full IPC access to the users; <literal>root</literal>,<literal>joe</literal>
+and to the members of the group <literal>wheel</literal>.</para>
```
Comment (dariavladykina) [inline xml/physical_security.xml:563]:
```suggestion
<para>The example allows full IPC access to the users <literal>root</literal>, <literal>joe</literal>
and to the members of the group <literal>wheel</literal>.</para>
```

File: xml/physical_security.xml [hunk 19]
```diff
@@ -426,4 +426,176 @@ polkit.addRule(function(action, subject) {
...
+</sect3>
+</sect2>
+<sect2 xml:id="sec-more-information-usbguard">
+<title>More information</title>
+<para>To know more about USBGuard, see: </para>
+<itemizedlist>
+  <listitem>
+    <para>
+      The upstream documentation at
+      <link xlink:href="https://usbguard.github.io/"/>.
```
Comment (dariavladykina) [inline xml/physical_security.xml:574]:
```suggestion
      <link xlink:href="https://usbguard.github.io/"/>
```

**Review by dariavladykina — COMMENTED** [review]
Hi Amrita, please see some suggestions here. Thanks!

File: xml/physical_security.xml (hunk 15 above)
Comment (Amrita42) [reply xml/physical_security.xml:488]: The main section is about how to use USBGuard  and that is via the _configuration file_ where its explained the various options that a user can use . I prefer to retain as is.
